# Bellabeat Case Study

## 1. Ask — Business Task

### Business Task

Analyze smart device usage data to identify trends in users' daily activity, sleep, and health-related behaviors. The goal is to understand how these insights could be applied to Bellabeat customers and support the company's marketing strategy.

### Key Questions

1. What are the main trends in smart device usage?
2. How do activity and sleep patterns vary among users?
3. How could these insights help Bellabeat improve its marketing strategy?

## 2. Data Sources

The analysis uses the Fitbit Fitness Tracker Data dataset.

The following datasets were selected:
- `dailyActivity_merged.csv`
- `sleepDay_merged.csv`

## 3. Data Cleaning

In [4]:
%pip install pandas

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 25.0.1 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [1]:
import pandas as pd

In [4]:
activity = pd.read_csv(
    r"C:\Users\UserComp\Data_Analytics\Bellabeat\data\dailyActivity_merged.csv"
)

sleep = pd.read_csv(
    r"C:\Users\UserComp\Data_Analytics\Bellabeat\data\sleepDay_merged.csv"
)

In [5]:
activity.head()

,Id,ActivityDate,TotalSteps,TotalDistance,TrackerDistance,LoggedActivitiesDistance,VeryActiveDistance,ModeratelyActiveDistance,LightActiveDistance,SedentaryActiveDistance,VeryActiveMinutes,FairlyActiveMinutes,LightlyActiveMinutes,SedentaryMinutes,Calories
0,1503960366,3/25/2016,11004,7.11,7.11,0.0,2.57,0.46,4.07,0.0,33,12,205,804,1819
1,1503960366,3/26/2016,17609,11.55,11.55,0.0,6.92,0.73,3.91,0.0,89,17,274,588,2154
2,1503960366,3/27/2016,12736,8.53,8.53,0.0,4.66,0.16,3.71,0.0,56,5,268,605,1944
3,1503960366,3/28/2016,13231,8.93,8.93,0.0,3.19,0.79,4.95,0.0,39,20,224,1080,1932
4,1503960366,3/29/2016,12041,7.85,7.85,0.0,2.16,1.09,4.61,0.0,28,28,243,763,1886


In [6]:
sleep.head()

,Id,SleepDay,TotalSleepRecords,TotalMinutesAsleep,TotalTimeInBed
0,1503960366,4/12/2016 12:00:00 AM,1,327,346
1,1503960366,4/13/2016 12:00:00 AM,2,384,407
2,1503960366,4/15/2016 12:00:00 AM,1,412,442
3,1503960366,4/16/2016 12:00:00 AM,2,340,367
4,1503960366,4/17/2016 12:00:00 AM,1,700,712


In [7]:
activity.info()

<class 'pandas.DataFrame'>
RangeIndex: 457 entries, 0 to 456
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        457 non-null    int64  
 1   ActivityDate              457 non-null    str    
 2   TotalSteps                457 non-null    int64  
 3   TotalDistance             457 non-null    float64
 4   TrackerDistance           457 non-null    float64
 5   LoggedActivitiesDistance  457 non-null    float64
 6   VeryActiveDistance        457 non-null    float64
 7   ModeratelyActiveDistance  457 non-null    float64
 8   LightActiveDistance       457 non-null    float64
 9   SedentaryActiveDistance   457 non-null    float64
 10  VeryActiveMinutes         457 non-null    int64  
 11  FairlyActiveMinutes       457 non-null    int64  
 12  LightlyActiveMinutes      457 non-null    int64  
 13  SedentaryMinutes          457 non-null    int64  
 14  Calories             

In [8]:
sleep.info()

<class 'pandas.DataFrame'>
RangeIndex: 413 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   Id                  413 non-null    int64
 1   SleepDay            413 non-null    str  
 2   TotalSleepRecords   413 non-null    int64
 3   TotalMinutesAsleep  413 non-null    int64
 4   TotalTimeInBed      413 non-null    int64
dtypes: int64(4), str(1)
memory usage: 16.3 KB


In [9]:
activity.isnull().sum()

Id                          0
ActivityDate                0
TotalSteps                  0
TotalDistance               0
TrackerDistance             0
LoggedActivitiesDistance    0
VeryActiveDistance          0
ModeratelyActiveDistance    0
LightActiveDistance         0
SedentaryActiveDistance     0
VeryActiveMinutes           0
FairlyActiveMinutes         0
LightlyActiveMinutes        0
SedentaryMinutes            0
Calories                    0
dtype: int64

In [10]:
sleep.isnull().sum()

Id                    0
SleepDay              0
TotalSleepRecords     0
TotalMinutesAsleep    0
TotalTimeInBed        0
dtype: int64

In [11]:
activity.duplicated().sum()

np.int64(0)

In [12]:
sleep.duplicated().sum()

np.int64(3)

In [13]:
sleep[sleep.duplicated()]

,Id,SleepDay,TotalSleepRecords,TotalMinutesAsleep,TotalTimeInBed
161,4388161847,5/5/2016 12:00:00 AM,1,471,495
223,4702921684,5/7/2016 12:00:00 AM,1,520,543
380,8378563200,4/25/2016 12:00:00 AM,1,388,402


In [14]:
sleep.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
408    False
409    False
410    False
411    False
412    False
Length: 413, dtype: bool

In [15]:
sleep[sleep.duplicated()]

,Id,SleepDay,TotalSleepRecords,TotalMinutesAsleep,TotalTimeInBed
161,4388161847,5/5/2016 12:00:00 AM,1,471,495
223,4702921684,5/7/2016 12:00:00 AM,1,520,543
380,8378563200,4/25/2016 12:00:00 AM,1,388,402


In [16]:
sleep = sleep.drop_duplicates()

In [17]:
sleep.duplicated().sum()

np.int64(0)

In [18]:
activity.dtypes

Id                            int64
ActivityDate                    str
TotalSteps                    int64
TotalDistance               float64
TrackerDistance             float64
LoggedActivitiesDistance    float64
VeryActiveDistance          float64
ModeratelyActiveDistance    float64
LightActiveDistance         float64
SedentaryActiveDistance     float64
VeryActiveMinutes             int64
FairlyActiveMinutes           int64
LightlyActiveMinutes          int64
SedentaryMinutes              int64
Calories                      int64
dtype: object

In [19]:
activity["ActivityDate"] = pd.to_datetime(activity["ActivityDate"])

In [20]:
activity.dtypes

Id                                   int64
ActivityDate                datetime64[us]
TotalSteps                           int64
TotalDistance                      float64
TrackerDistance                    float64
LoggedActivitiesDistance           float64
VeryActiveDistance                 float64
ModeratelyActiveDistance           float64
LightActiveDistance                float64
SedentaryActiveDistance            float64
VeryActiveMinutes                    int64
FairlyActiveMinutes                  int64
LightlyActiveMinutes                 int64
SedentaryMinutes                     int64
Calories                             int64
dtype: object

In [23]:
sleep["SleepDay"] = pd.to_datetime(
    sleep["SleepDay"],
    format="%m/%d/%Y %I:%M:%S %p"
)

In [24]:
sleep.dtypes

Id                             int64
SleepDay              datetime64[us]
TotalSleepRecords              int64
TotalMinutesAsleep             int64
TotalTimeInBed                 int64
dtype: object

In [25]:
activity["Id"].nunique()

35

In [26]:
sleep["Id"].nunique()

24

In [27]:
activity["ActivityDate"].min(), activity["ActivityDate"].max()

(Timestamp('2016-03-12 00:00:00'), Timestamp('2016-04-12 00:00:00'))

In [28]:
sleep["SleepDay"].min(), sleep["SleepDay"].max()

(Timestamp('2016-04-12 00:00:00'), Timestamp('2016-05-12 00:00:00'))

### Data Cleaning Summary

I checked the datasets for missing values, duplicate records, and data types.

- No missing values were found.
- No duplicate records were found in the activity dataset.
- Three duplicate records were found and removed from the sleep dataset.
- Date columns were converted from text to datetime format.
- The activity dataset contains 35 unique users.
- The sleep dataset contains 24 unique users.
- The activity and sleep datasets cover different time periods.

## 4. Analyze — Data Analysis

In [29]:
activity.describe()

,Id,ActivityDate,TotalSteps,TotalDistance,TrackerDistance,LoggedActivitiesDistance,VeryActiveDistance,ModeratelyActiveDistance,LightActiveDistance,SedentaryActiveDistance,VeryActiveMinutes,FairlyActiveMinutes,LightlyActiveMinutes,SedentaryMinutes,Calories
count,4.570000e+02,457,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000
mean,4.628595e+09,2016-04-04 04:40:26.258205,6546.562363,4.663523,4.609847,0.179427,1.180897,0.478643,2.890197,0.001904,16.623632,13.070022,170.070022,995.282276,2189.452954
min,1.503960e+09,2016-03-12 00:00:00,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,32.000000,0.000000
25%,2.347168e+09,2016-04-02 00:00:00,1988.000000,1.410000,1.280000,0.000000,0.000000,0.000000,0.870000,0.000000,0.000000,0.000000,64.000000,728.000000,1776.000000
50%,4.057193e+09,2016-04-05 00:00:00,5986.000000,4.090000,4.090000,0.000000,0.000000,0.020000,2.930000,0.000000,0.000000,1.000000,181.000000,1057.000000,2062.000000
75%,6.391747e+09,2016-04-08 00:00:00,10198.000000,7.160000,7.110000,0.000000,1.310000,0.670000,4.460000,0.000000,25.000000,16.000000,257.000000,1285.000000,2667.000000
max,8.877689e+09,2016-04-12 00:00:00,28497.000000,27.530001,27.530001,6.727057,21.920000,6.400000,12.510000,0.100000,202.000000,660.000000,720.000000,1440.000000,4562.000000
std,2.293781e+09,NaN,5398.493064,4.082072,4.068540,0.849232,2.487159,0.830995,2.237523,0.008487,28.919704,36.208635,122.205372,337.021404,815.484523


In [30]:
activity[[
    "TotalSteps",
    "TotalDistance",
    "SedentaryMinutes",
    "Calories"
]].mean()

TotalSteps          6546.562363
TotalDistance          4.663523
SedentaryMinutes     995.282276
Calories            2189.452954
dtype: float64

In [31]:
activity[[
    "VeryActiveMinutes",
    "FairlyActiveMinutes",
    "LightlyActiveMinutes",
    "SedentaryMinutes"
]].mean()

VeryActiveMinutes        16.623632
FairlyActiveMinutes      13.070022
LightlyActiveMinutes    170.070022
SedentaryMinutes        995.282276
dtype: float64

In [32]:
activity[[
    "VeryActiveMinutes",
    "FairlyActiveMinutes",
    "LightlyActiveMinutes",
    "SedentaryMinutes"
]].mean().sort_values()

FairlyActiveMinutes      13.070022
VeryActiveMinutes        16.623632
LightlyActiveMinutes    170.070022
SedentaryMinutes        995.282276
dtype: float64

In [33]:
activity["TotalSteps"].mean()

np.float64(6546.562363238512)

In [34]:
(activity["TotalSteps"] >= 10000).mean() * 100

np.float64(27.78993435448578)

In [35]:
sleep["TotalMinutesAsleep"].mean()

np.float64(419.17317073170733)

In [36]:
(sleep["TotalMinutesAsleep"] < 420).mean() * 100

np.float64(44.146341463414636)

In [37]:
activity["ActivityDate"].min(), activity["ActivityDate"].max()

(Timestamp('2016-03-12 00:00:00'), Timestamp('2016-04-12 00:00:00'))

In [38]:
sleep["SleepDay"].min(), sleep["SleepDay"].max()

(Timestamp('2016-04-12 00:00:00'), Timestamp('2016-05-12 00:00:00'))

In [39]:
set(activity["ActivityDate"]).intersection(set(sleep["SleepDay"]))

{Timestamp('2016-04-12 00:00:00')}

In [40]:
sleep["TotalTimeInBed"].mean()

np.float64(458.4829268292683)

In [41]:
sleep["TimeAwakeInBed"] = (
    sleep["TotalTimeInBed"] - sleep["TotalMinutesAsleep"]
)

sleep["TimeAwakeInBed"].mean()

np.float64(39.30975609756098)

In [42]:
activity["DayOfWeek"] = activity["ActivityDate"].dt.day_name()

In [43]:
activity[["ActivityDate", "DayOfWeek"]].head()

,ActivityDate,DayOfWeek
0,2016-03-25,Friday
1,2016-03-26,Saturday
2,2016-03-27,Sunday
3,2016-03-28,Monday
4,2016-03-29,Tuesday


In [44]:
activity.groupby("DayOfWeek")["TotalSteps"].mean()

DayOfWeek
Friday       6737.561644
Monday       7118.588235
Saturday     7089.773333
Sunday       6058.013889
Thursday     6847.083333
Tuesday      4914.917808
Wednesday    7510.708333
Name: TotalSteps, dtype: float64

In [45]:
activity["DayOfWeek"].value_counts()

DayOfWeek
Saturday     75
Friday       73
Tuesday      73
Sunday       72
Monday       68
Wednesday    48
Thursday     48
Name: count, dtype: int64

In [46]:
activity.groupby("DayOfWeek")["Id"].nunique()

DayOfWeek
Friday       35
Monday       35
Saturday     35
Sunday       35
Thursday     35
Tuesday      35
Wednesday    35
Name: Id, dtype: int64

In [47]:
user_steps = activity.groupby("Id")["TotalSteps"].mean()
user_steps

Id
1503960366    11640.526316
1624580081     4226.263158
1644430081     9274.800000
1844505072     3640.583333
1927972279     2180.833333
2022484408    12174.916667
2026352035     3392.750000
2320127002     3138.416667
2347167796     9800.066667
2873212765     6636.583333
2891001357      773.625000
3372868164     6127.500000
3977333714     8663.916667
4020332650     5776.593750
4057192912     1887.218750
4319703577     7820.583333
4388161847        0.000000
4445114986     4293.066667
4558609924     5785.250000
4702921684     7943.400000
5553957443     8354.750000
5577150313     8608.181818
6117666160     8248.600000
6290855005     1618.100000
6391747486     1336.888889
6775888955     5559.000000
6962181067    12639.714286
7007744171    12260.333333
7086361926     6103.916667
8053475328    14844.363636
8253242879     2389.916667
8378563200     8135.250000
8583815059     3045.500000
8792009665     3094.916667
8877689391    17417.083333
Name: TotalSteps, dtype: float64

In [48]:
def activity_level(steps):
    if steps < 5000:
        return "Low"
    elif steps < 10000:
        return "Moderate"
    else:
        return "High"

user_activity = user_steps.apply(activity_level)

user_activity.value_counts()

TotalSteps
Moderate    15
Low         14
High         6
Name: count, dtype: int64

### Data Analysis Key Findings

- The average number of steps was 6,547 per day.
- Users recorded 10,000 or more steps on 27.8% of days.
- 14 users had low average activity, 15 had moderate activity, and 6 had high activity.
- The average recorded sedentary time was about 995 minutes per day.
- The average sleep duration was about 7 hours per day.
- 44.1% of recorded sleep days had less than 7 hours of sleep.

## 5. Share — Data Visualization

In [51]:
activity_summary = user_activity.value_counts().reset_index()

activity_summary.columns = ["ActivityLevel", "Users"]

activity_summary

,ActivityLevel,Users
0,Moderate,15
1,Low,14
2,High,6


In [52]:
activity_summary.to_csv(
    r"C:\Users\UserComp\Data_Analytics\Bellabeat\data\activity_summary.csv",
    index=False
)

In [53]:
activity_minutes = activity[[
    "VeryActiveMinutes",
    "FairlyActiveMinutes",
    "LightlyActiveMinutes",
    "SedentaryMinutes"
]].mean()

activity_minutes

VeryActiveMinutes        16.623632
FairlyActiveMinutes      13.070022
LightlyActiveMinutes    170.070022
SedentaryMinutes        995.282276
dtype: float64

In [54]:
activity_minutes_summary = activity_minutes.reset_index()

activity_minutes_summary.columns = ["ActivityType", "Minutes"]

activity_minutes_summary

,ActivityType,Minutes
0,VeryActiveMinutes,16.623632
1,FairlyActiveMinutes,13.070022
2,LightlyActiveMinutes,170.070022
3,SedentaryMinutes,995.282276


In [55]:
activity_minutes_summary.to_csv(
    r"C:\Users\UserComp\Data_Analytics\Bellabeat\data\activity_minutes_summary.csv",
    index=False
)

In [56]:
sleep_summary = sleep[
    ["TotalMinutesAsleep", "TotalTimeInBed"]
].mean()

sleep_summary

TotalMinutesAsleep    419.173171
TotalTimeInBed        458.482927
dtype: float64

sleep_summary_table = sleep_summary.reset_index()

sleep_summary_table.columns = ["SleepMetric", "Minutes"]

sleep_summary_table

In [58]:
sleep_summary_table.to_csv(
    r"C:\Users\UserComp\Data_Analytics\Bellabeat\data\sleep_summary.csv",
    index=False
)

## 6. Act - Recommendations


### Final Conclusion

The analysis showed that most users had low or moderate activity levels. Only 6 out of 35 users had a high average activity level. The activity data also showed a large amount of sedentary time compared with active minutes.

Sleep was another important finding. Users slept for about 7 hours on average and spent about 7 hours and 38 minutes in bed. Also, 44.1% of recorded sleep days had less than 7 hours of sleep.

Overall, the data show that users could benefit from more daily movement and better sleep habits. Bellabeat could use these insights to encourage users to be more active and support healthier sleep routines through its app.

### Top 3 Recommendations

1. **Encourage daily movement**  
   Bellabeat could use the app to set simple daily step goals and send reminders to users who are less active.

2. **Promote better sleep habits**  
   Bellabeat could provide simple sleep tips and reminders to help users maintain a regular sleep schedule.

3. **Create personalized daily goals**  
   Bellabeat could use activity and sleep data to give users simple daily goals based on their habits.